stokespy/stokespy.py

```
@@ -31,94 +33,182 @@ class StokesParamCube(ndcube.ndcube.NDCubeBase):
     
-    def plot(self, wavelength=None, coord1=None, coord2=None):
+    def _get_index(self,wavelength):
+        """Test if a wavelength is inside the wavelength axis for the object and return the array index corresponding to that wavelength """
         
-        if wavelength is None:
-            # Need a better way to select which wavelength to show.
-            ix = int(self._spectral_axis.shape[0]/2)
+        # Test if the value is either an integer index or astropy quantity
+        if isinstance(wavelength, int):
+            ix = wavelength
+            if (ix < 0) or (ix > self.n_spectral-1):
+                ix = 0 if ix < 0 else (self.n_spectral-1)
+                print('Warning: Wavelength selected outside of axis range: {} {}'.\
+                        format(self._spectral_axis[0], self._spectral_axis[-1]))
+                print('Defaulting to nearest wavelength at {}'.\
+                        format(self._spectral_axis[ix]))
+                
+        elif isinstance(wavelength,u.Quantity):
+            # Unit check the input wavelength
+            wav = wavelength.to(self._spectral_slice().world_axis_units[0])
+            ix = int(self._spectral_slice().world_to_array_index_values((wav.value,))[0])
 
-        # Test if the wavelength provided is a Quantity object.
-        if isinstance(wavelength,astropy.units.Quantity):
-            ix = int(self._spectral_slice().world_to_array_index_values(wavelength))
-            # Check that the selected value falls within the wavelength array.
+            # Check that the wavelength is inside the cube spectral
+            # axis range.
             if (ix < 0) or (ix > self.n_spectral-1):
                 ix = 0 if ix < 0 else (self.n_spectral-1)
-                print('Warning: Wavelength selected outside of range: {} {}'.\
-                      format(self._spectral_axis[0], self._spectral_axis[-1]))
+                print('Warning: Wavelength selected outside of axis range: {} {}'.\
```

Prefer code to fail here instead of warning/fixing bad input.  This implementation could lead to unintended consequences for badly behaving, unmonitored pipelines.  Will accept now and make an issue for it.